### PR TITLE
Adds metadata home page in react app

### PR DESCRIPTION
**Pull request by antony-britto**
1) Add top level routing for metadata pages
2) Adds metadata home page component
3) Adds a feature flag and handles redirection between angular & react pages.

### CODE REVIEW

File: app/cdap/components/Metadata/index.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  function checkFeatureFlag(componentToRender: JSX.Element, urlToRedirect: string) {
+    if (!Theme.isMetadataInReact) {
+      window.location.href = urlToRedirect;
+    } else {
+      return componentToRender;
+    }
+  }
+
+  return (
+    <React.Fragment>
```
Comment (seanfeld) [inline app/cdap/components/Metadata/index.tsx]: Lets just use the `<></>` shorthand for react fragments.

File: app/cdap/services/ThemeHelper.ts [hunk 2]
```diff
@@ -223,6 +224,7 @@ interface IThemeObj {
   featureNames?: IFeatureNames;
   showSqlPipeline?: boolean;
   showCDC?: boolean;
+  isMetadataInReact?: boolean;
```
Comment (seanfeld) [inline app/cdap/services/ThemeHelper.ts:227]: Can we add a todo here? Eventually we want the feature flags to be in a context provider situation and I don't think keeping it in themes makes sense long term.

File: app/cdap/components/Metadata/Home/index.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+const MetadataHome: React.FC = () => {
+  const searchQueryRegex = new RegExp('^(?![*]).*$');
+  const [error, setError] = useState(false);
+
+  const handleSearch = (event) => {
+    const searchQuery = event.target.value;
+    const isValid = searchQueryRegex.test(searchQuery);
+    setError(!isValid);
+    if (event.keyCode === 13 && searchQuery.trim() !== '' && isValid) {
```
Comment (seanfeld) [inline app/cdap/components/Metadata/Home/index.tsx]: I think we can do `event.key === 'Enter'` or `event.code === 'Enter'` - i think its much easier/clearer than keycode.
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

Comment (antony-britto) [reply app/cdap/components/Metadata/Home/index.tsx]: Done

**Review by seanfeld — APPROVED** [review]

**Pull request merged**